Functions decomposition

ccrvlh opened the pull request:
The idea here is to make functions smaller, and easier to comprehend (and maintain).
I tried to split the big functions (mainly the main cli, operations and operation) into smaller chunks that made sense (not sure I was able to). I tried to follow the logic already present using the comment blocks.

Example for the main:
```
_setup_log_level()
_validate_operations()
_setup_state()
_set_override_data()
_set_config()
_set_verbosity()
_apply_inventory_limit()
```

Unfortunately this made the full diff quite a pain specially for the cli part, I’m sorry about that. Hopefully looking at the code directly will make more sense. The good part is that it made the function a lot smaller (from ~350 to less than 100 lines), and easier to understand (in my opinion, of course).

Also tried to slightly reduce the indented and nested parts of the code (this usually happens with log messages) - as a first suggestion, I chose to go to the “more variables, less indentation path” but this is purely a personal choice (I personally feel is easier to read & follow). A lot of these are local/private, and won’t be used anywhere else, hence the _ prefix.
In general I chose to play safe, sometimes returning more information that what’s necessary, mainly because there are a few places that change global vars, so I though it would be better to return more than what’s necessary.

I made an effort not to change any piece of logic, just purely organizing functions. Tests are passing normally. I hope you guys like it, but I understand there’s a lot of personal preferences here, let me now what you think and I’ll be happy to adjust accordingly. If none of this makes sense, I’ll also understand, no worries at all.

Edit: the last PR was hard to read because of the diff. Couldn't really make the diff better, but I split the commits a lot more, so the diff in each commit should be a lot easier to read.

ccrvlh committed:
Log Level, Validation Ops, Setup State, Config

ccrvlh committed:
Inventory Limit & Operation Handlers

ccrvlh committed:
Echo Msg

ccrvlh committed:
Operation: Attach Args

ccrvlh committed:
Operation: execute_immediately

ccrvlh committed:
Operation: Solv Legacy Op Args

ccrvlh committed:
Operation: Generate Name

ccrvlh committed:
Operation: Solve Consistency

ccrvlh committed:
Operation: _ensure_shared_op_meta & Func Order

ccrvlh committed:
Operation: Update State Meta

ccrvlh committed:
Operations reduce log nesting

Fizzadar commented on pyinfra/api/operations.py:
```diff
@@ -128,18 +128,12 @@ def run_host_op(state, host, op_hash):
+                _error_msg_styled = click.style(_error_msg, "red")
+                _error_log = "{0}{1}".format(host.print_prefix, _error_msg_styled)
                 logger.warning(traceback.format_exc())
-                logger.error(
```
Heh - this is such a huge improvement to readability, the nesting here makes my eyes burn 😆

ccrvlh replied:
Yeah, it was a bit confusing with a lot of nesting. I don't know if this is the best solution, but at least it's "flat", a bit of an improvement I would say!

Fizzadar commented on pyinfra/api/operation.py:
```diff
@@ -222,30 +219,9 @@ def decorated_func(*args, **kwargs):
-        # If we're already in the execution phase, execute this operation immediately
-        if state.is_executing:
-            _execute_immediately(state, host, op_data, op_meta, op_hash)
+        state, operation_meta = _update_state_meta(
```
This needn't return `state` because it mutates it, which I think we should also add a commend indicating.

Fizzadar replied:
https://github.com/Fizzadar/pyinfra/pull/883/commits/f30aa877f7cdb5349d0f41f3459886c3255532af

Fizzadar commented on pyinfra/api/operation.py:
```diff
@@ -306,50 +331,29 @@ def _solve_operation_consistency(names, state, host):
-        names = {"{0} | {1}".format(host.current_deploy_name, name) for name in names}
-
-    return names, add_args
+def _ensure_shared_op_meta(state, op_hash, op_order, global_kwargs, names):
```
Let's add a comment indicating this function modifies state.

Fizzadar replied:
https://github.com/Fizzadar/pyinfra/pull/883/commits/f30aa877f7cdb5349d0f41f3459886c3255532af

Fizzadar reviewed (approved):
This looks great @lowercase00, thank you for doing this it makes things much more approachable. Left a few comments for myself mainly I'll add them in before merging this hopefully this weekend!

Fizzadar committed:
Merge branch '2.x' into new_split

# Conflicts:
#	pyinfra/api/operation.py

Fizzadar committed:
Drop `flake8-commas`

Black already enforces styling and conflicts with trailing commas
on a function call with dedicated arguments line. Regularly catches
people out and doesn't yield any real benefit anymore.

Fizzadar committed:
Add comments in `operation.py`

Fizzadar committed:
Typo

ccrvlh commented:
@Fizzadar let me know if there's anything else to change in here

Fizzadar commented:
All good @lowercase00 I'm just behind on things! Thank you for working on this, an excellent improvement 👍!
